introspection-engine/connectors/sql-introspection-connector/tests/test_harness/misc_helpers.rs

```diff
@@ -0,0 +1,311 @@
+use crate::test_harness::{IntrospectionDatabase, IntrospectionDatabaseWrapper, Mysql, PostgreSql, Sqlite};
+use barrel::{Migration, SqlVariant};
+use introspection_connector::IntrospectionConnector;
+use pretty_assertions::assert_eq;
+use prisma_query::connector::{MysqlParams, PostgresParams};
+use sql_introspection_connector::*;
+use std::{convert::TryFrom, rc::Rc, sync::Arc};
+use url::Url;
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum SqlFamily {
+    Sqlite,
+    Postgres,
+    Mysql,
+}
+
+pub struct TestSetup {
+    pub sql_family: SqlFamily,
+    pub database: Arc<dyn IntrospectionDatabase + Send + Sync + 'static>,
+    pub introspection_connector: Box<dyn IntrospectionConnector>,
+}
+
+pub struct BarrelMigrationExecutor {
+    database: Arc<dyn IntrospectionDatabase + Send + Sync>,
+    sql_variant: barrel::backend::SqlVariant,
+}
+
+// test execution
+
+pub(crate) fn custom_assert(left: String, right: String) {
+    let parsed_expected = datamodel::parse_datamodel(&right).unwrap();
+    let reformatted_expected =
+        datamodel::render_datamodel_to_string(&parsed_expected).expect("Datamodel rendering failed");
+
+    assert_eq!(left, reformatted_expected);
+}
```

Comment:
very nitpicky, but this function doesn't need to take ownership. `left` and `right` could both be `&str`. The references can make `String`s and `&str`s annoying to compare, but you can always go from a `String` or `&String` to a `&str` with `.as_str()`.